src/reader/location.rs

```diff
@@ -0,0 +1,206 @@
+/// Parsers for location specifiers
+use std::str;
+
+use branch::alt;
+use bytes::complete::{is_not, tag};
+use character::complete::{char, i64};
+use combinator::{map, map_opt, map_res, opt};
+use multi::separated_list1;
+use nom::combinator::value;
+use nom::{IResult, Parser};
+use nom::OutputM;
+use nom::Check;
+use nom::{branch, bytes, character, combinator, multi, sequence};
+use sequence::{preceded, separated_pair};
+
+use crate::seq::{After, Before, GapLength, Location};
+
+fn location_single(input: &[u8]) -> IResult<&[u8], Location> {
+    map(i64, |i| {
+        Location::Range(((i - 1), Before(false)), (i, After(false)))
+    }).parse(input)
+}
+
+fn location_between(input: &[u8]) -> IResult<&[u8], Location> {
+    map_opt(
+        separated_pair(i64, char('^'), i64),
+        // This check is not 100% foolproof, because if b == 1, a must be
+        // the last nt of a circular sequence, but since we might not yet
+        // know the sequence length, we can't check this yet
+        |(a, b)| {
+            ((a - b).abs() == 1 || ((a == 1) ^ (b == 1))).then_some(Location::Between(a - 1, b - 1))
+        },
+    ).parse(input)
+}
+
+fn location_span(input: &[u8]) -> IResult<&[u8], Location> {
+    let parser = (opt(char('<')), i64, tag(".."), opt(char('>')), i64);
+    map(parser, |(before, a, _, after, b)| {
+        Location::Range(
+            (a - 1, Before(before.is_some())),
+            (b, After(after.is_some())),
+        )
+    }).parse(input)
+}
+
+fn operator<'a, F, O>(name: &'static str, inner: F) -> impl Parser<&'a [u8], Output = O, Error = nom::error::Error<&'a [u8]>> 
+where
+    F: Parser<&'a [u8], Output = O, Error = nom::error::Error<&'a [u8]>>
+{
```

Comment:
Would `map((tag(name), char('('), inner, char(')')), |(_, _, output, _)| output)` work here?

Reply:
Actually I replaced it with:
```rust
terminated(
    preceded(tag(name), preceded(char('('), inner)),
    char(')'),
)
```

The problem with using `map` with some ignored arguments is that the parsers must still produce a value for them, which cannot benefit from the new `Check` mode that now makes a parser faster to use when it's just checking in v8.